test/src/tests-forwarders.js

```diff
@@ -11,65 +11,60 @@ import {
 } from './config/constants';
 import { expect } from 'chai';
 
-const findEventFromRequest = Utils.findEventFromRequest,
-    getForwarderEvent = Utils.getForwarderEvent,
-    setLocalStorage = Utils.setLocalStorage,
-    forwarderDefaultConfiguration = Utils.forwarderDefaultConfiguration,
-    MockForwarder = Utils.MockForwarder,
-    MockSideloadedKit = Utils.MockSideloadedKit;
-let mockServer;
-
-describe('forwarders', function() {
+const { findEventFromRequest,
+    waitForCondition,
+    fetchMockSuccess,
+    getForwarderEvent,
+    getIdentityEvent,
+    setLocalStorage,
+    forwarderDefaultConfiguration,
+    MockForwarder,
+    MockSideloadedKit,hasIdentifyReturned
+ } = Utils;
+
+ let mockServer;
+
+describe.only('forwarders', function() {
     beforeEach(function() {
         mParticle._resetForTests(MPConfig);
         delete mParticle._instances['default_instance'];
         fetchMock.post(urls.events, 200);
         mockServer = sinon.createFakeServer();
-        mockServer.respondImmediately = true;
 
-        mockServer.respondWith(urls.identify, [
-            200,
-            {},
-            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
-        ]);
+        fetchMockSuccess(urls.identify, {
+            mpid: testMPID,
+            is_logged_in: false,
+        });
 
-        mockServer.respondWith(urls.login, [
-            200,
-            {},
-            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
-        ]);
+        fetchMockSuccess(urls.login, {
+            mpid: testMPID, is_logged_in: false
+        });
 
-        mockServer.respondWith(urls.logout, [
-            200,
-            {},
-            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
-        ]);
+        fetchMockSuccess(urls.logout, {
+            mpid: testMPID, is_logged_in: true
+        })
 
-        mockServer.respondWith(urls.modify, [
-            200,
-            {},
-            JSON.stringify({
-                change_results: [
+        fetchMockSuccess(urls.modify, {
+            change_results: [
                     {
                         identity_type: 'email',
                         modified_mpid: testMPID,
                     },
                 ],
-            }),
-        ]);
+        })
+
+        fetchMockSuccess(urls.forwarding, { mpid: testMPID, is_logged_in: false })
 
         mockServer.respondWith(urls.forwarding, [
             202,
             {},
-            JSON.stringify({ mpid: testMPID, is_logged_in: false }),
+            JSON.stringify({}),
         ]);
-        mParticle.init(apiKey, window.mParticle.config);
```

Comment:
Why is `init` being removed?

Reply:
I found that it was interfering with a lot of tests.  If we instead remove the init from the before blocks, and then call them only when needed, it reduces the number of `resetForTests` we need elsewhere and keeps each test a bit more self contained with fewer side effects.